geps/gep-3388/index.md

```diff
@@ -79,13 +79,265 @@ The implementation of a version of Linkerd's `ttl` parameter within Envoy might
+  //
+  // Support: Extended
+  //
+  // +optional
+  // <gateway:experimental>
+  Retry *CommonRetryPolicy `json:"retry,omitempty"`
+
+  // SessionPersistence defines and configures session persistence
+  // for the backend.
+  //
+  // Support: Extended
+  //
+  // +optional
+  SessionPersistence *SessionPersistence `json:"sessionPersistence,omitempty"`
+}
+
+// CommonRetryPolicy defines the configuration for when to retry a request.
+//
+type CommonRetryPolicy struct {
+    // TODO: Does it make sense to include this configuration in the policy or not?
+    //
+    // Support: Extended
+    //
+    // +optional
+    HTTP *HTTPRouteRetry `json:"http,omitempty"`
+
+    // Support: Extended
+    //
+    // +optional
+    BudgetPercent *Int `json:"budgetPercent,omitempty"`
+
+    // Support: Extended
+    //
+    // +optional
+    BudgetInterval *Duration `json:"budgetInterval,omitempty"`
+
+    // Support: Extended
+    //
+    // +optional
+    minRetryRate *RequestRate `json:"retryRate,omitempty"`
+}
+
+// RequestRate expresses a rate of requests over a given period of time.
+//
+type RequestRate struct {
+    // Support: Extended
+    //
+    // +optional
+    Count *Int `json:"count,omitempty"`
+
+    // Support: Extended
+    //
+    // +optional
+    Interval *Duration `json:"interval,omitempty"`
+}
+
+// Duration is a string value representing a duration in time. The foramat is
+// as specified in GEP-2257, a strict subset of the syntax parsed by Golang
+// time.ParseDuration.
+//
+// +kubebuilder:validation:Pattern=`^([0-9]{1,5}(h|m|s|ms)){1,4}$`
+type Duration string
 
 ### YAML
 
-TODO
+```yaml
+apiVersion: gateway.networking.x-k8s.io/v1alpha1
+kind: RetryPolicy
+metadata:
+  name: retry-policy-example
+spec:
+  targetRefs:
+    - group: ""
+      kind: Service
+      name: foo
+  from:
+    - kind: Mesh
+      namespace: istio-system
+      name: istio
+    - kind: Gateway
+      name: foo-ingress
+  http:
+    codes:
+    - 500
+    - 502
+    - 503
+    - 504
+    attempts: 2
+    backoff: 100ms
+  budgetPercent: 20
+  budgetInterval: 10s
+  minRetryRate:
+    count: 3
+    interval: 1s
+status:
+  ancestors:
+  - ancestorRef:
+      kind: Mesh
+      namespace: istio-system
+      name: istio
+    controllerName: "istio.io/mesh-controller"
+    conditions:
+    - type: "Accepted"
+      status: "True"
+      reason: "Accepted"
+  - ancestorRef:
+      kind: Gateway
+      namespace: foo-ns
+      name: foo-ingress
+    controllerName: "istio.io/mesh-controller"
+    conditions:
+    - type: "Accepted"
+      status: "False"
+      reason: "Invalid"
+      message: "RetryPolicy fields budgetPercentage, budgetInterval and minRetryRate are not supported for Istio ingress gateways."
+```
+
+```yaml
+apiVersion: gateway.networking.x-k8s.io/v1alpha1
+kind: BackendTrafficPolicy
+metadata:
+  name: traffic-policy-example
+spec:
+  targetRefs:
+    - group: ""
+      kind: Service
+      name: foo
+  from:
```

Comment:
nit: I would not call it from, maybe `scope`? not great either

Reply:
Chose `from` as it would mirror similar terminology in both [ReferenceGrant](https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1beta1.ReferenceGrantSpec) and [Istio AuthorizationPolicy](https://istio.io/latest/docs/reference/config/security/authorization-policy/), but [considering removing this from the initial scope](https://github.com/kubernetes-sigs/gateway-api/pull/3573#discussion_r1936377450) now.